# Generated by Django 3.2.4 on 2021-09-08 23:47

from django.conf import settings
import django.contrib.gis.db.models.fields
import django.contrib.postgres.fields
from django.db import migrations, models
import django.db.models.deletion
import django.utils.timezone


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='Reporter',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('photo', models.ImageField(null=True, upload_to='')),
                ('user', models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.CreateModel(
            name='Report',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('address', models.CharField(default='', max_length=128)),
                ('comment', models.CharField(default='', max_length=1024)),
                ('date', models.DateTimeField(default=django.utils.timezone.now)),
                ('name', models.CharField(default='', max_length=32)),
                ('photos', django.contrib.postgres.fields.ArrayField(base_field=models.ImageField(upload_to=''), default=list, size=None)),
                ('place', django.contrib.gis.db.models.fields.PointField(db_index=True, srid=4326)),
                ('status', models.CharField(choices=[('RECEIVED', 'Received'), ('APPROVED', 'Approved'), ('INVALID', 'Invalid')], default='RECEIVED', max_length=8)),
                ('type', models.CharField(default='', max_length=64)),
                ('subs', models.ManyToManyField(to=settings.AUTH_USER_MODEL)),
            ],
        ),
    ]
